plugin-taskbar/lxqttaskbar.cpp

```diff
@@ -304,11 +305,33 @@ void LXQtTaskBar::addWindow(WId window)
         connect(group, &LXQtTaskButton::dragging, this, [this] (QObject * dragSource, QPoint const & pos) {
             buttonMove(qobject_cast<LXQtTaskGroup *>(sender()), qobject_cast<LXQtTaskGroup *>(dragSource), pos);
         });
-
         mLayout->addWidget(group);
         group->setToolButtonsStyle(mButtonStyle);
-    }
+        const QString window_class = QString::fromUtf8(KWindowInfo(window, NET::Properties(), NET::WM2WindowClass).windowClassClass());
+        int src_index = mLayout->count() - 1;
+        int dst_index = src_index;
+        if (mUngroupedNextToExisting)
+        {
+            for (int i = mLayout->count() - 2; 0 <= i; --i)
+            {
+                LXQtTaskGroup * group = qobject_cast<LXQtTaskGroup*>(mLayout->itemAt(i)->widget());
```

Comment:
The name `group` is already used outside your block. Although it isn't wrong to use it here, to avoid confusion, please give it another name, like `thisGroup`.